plugins/by-name/easy-dotnet/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, config, ... }:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "easy-dotnet";
+  packPathName = "easy-dotnet.nvim";
+  package = "easy-dotnet-nvim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  settingsExample = {
+    test_runner = {
+      enable_buffer_test_execution = true;
+      viewmode = "float";
+    };
+    auto_bootstrap_namespace = true;
+    terminal.__raw = ''
+      function(path, action, args)
+        local commands = {
+          run = function()
+            return string.format("dotnet run --project %s %s", path, args)
+          end,
+          test = function()
+            return string.format("dotnet test %s %s", path, args)
+          end,
+          restore = function()
+            return string.format("dotnet restore %s %s", path, args)
+          end,
+          build = function()
+            return string.format("dotnet build %s %s", path, args)
+          end,
+        }
+
+        local command = commands[action]() .. "\r"
+        require("toggleterm").exec(command, nil, nil, nil, "float")
+      end
+    '';
+    picker = "fzf";
+  };
+
+  extraConfig = cfg: {
+    warnings = lib.nixvim.mkWarnings "plugins.easy-dotnet" (
+      lib.mapAttrsToList
+        (pickerName: pluginName: {
+          when = (cfg.settings.picker or null == pickerName) && (!config.plugins.${pluginName}.enable);
```

Comment:
This doesn't offer any way to suppress the warning if end-users explicitly don't want to enable our plugins.

Maybe that's not too important, as it seems like a contrived scenario.

But we could do something like:
```nix
when =
  cfg.settings.picker or null == pickerName
  && !config.plugins.${pluginName}.enable
  && options.plugins.${pluginName}.enable.highestPrio == 1500;
```

And say:
```
You can suppress this warning by explicitly setting `plugins.${pluginName}.enable = false`.
```